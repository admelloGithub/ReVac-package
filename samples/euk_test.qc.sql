\set semicolon_hack=1
\go

--
-- featureloc fmin-fmax checks
--
select 'Q1: Checking featureloc records where fmin>fmax';

select count(*)
from featureloc
where fmin>fmax;


--
-- dangling feature_dbxref records
--
select 'Q2: Checking for dangling feature_dbxref records';

select count(fd.feature_id) 
from feature_dbxref fd 
where not exists (
select 1 
from feature f 
where f.feature_id = fd.feature_id );

--
-- dangling organism records
--
select 'Q3: Checking for dangling organism records';

select o.genus, o.species, count(o.organism_id) 
from organism o 
where not exists (
select 1 
from feature f 
where f.organism_id = o.organism_id)
group by o.genus, o.species
having count(o.organism_id) > 0;
