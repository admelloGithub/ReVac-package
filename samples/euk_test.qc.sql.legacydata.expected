                                                
 -----------------------------------------------
 Q1: Checking featureloc records where fmin>fmax

(1 row affected)
            
 -----------
           0

(1 row affected)
                                                 
 ------------------------------------------------
 Q2: Checking for dangling feature_dbxref records

(1 row affected)
            
 -----------
           0

(1 row affected)
                                           
 ------------------------------------------
 Q3: Checking for dangling organism records

(1 row affected)
 genus                                             
	species                                                       
 --------------------------------------------------
	-------------------------------------------------- -----------

(0 rows affected)
